pkg/ipam-controller/controllers/configmap/configmap.go

```diff
@@ -0,0 +1,132 @@
+/*
+ Copyright 2023, NVIDIA CORPORATION & AFFILIATES
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+     http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net"
+
+	corev1 "k8s.io/api/core/v1"
+	apiErrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/Mellanox/nvidia-k8s-ipam/pkg/ipam-controller/allocator"
+	"github.com/Mellanox/nvidia-k8s-ipam/pkg/ipam-controller/config"
+	"github.com/Mellanox/nvidia-k8s-ipam/pkg/ipam-controller/selector"
+)
+
+// ConfigMapReconciler reconciles ConfigMap objects
+type ConfigMapReconciler struct {
+	Allocator          *allocator.Allocator
+	Selector           *selector.Selector
+	ConfigMapName      string
+	ConfigMapNamespace string
+
+	ConfigEventCh chan event.GenericEvent
+	client.Client
+	Scheme *runtime.Scheme
+}
+
+// Reconcile contains logic to sync Node objects
+func (r *ConfigMapReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	reqLog := log.FromContext(ctx)
+	if req.Name != r.ConfigMapName || req.Namespace != r.ConfigMapNamespace {
+		// this should never happen because of the watcher configuration
```

Comment:
thats because of the cache configuration in manager right  ?

perhaps add suffix: "in controller manager" ?

Reply:
yes, I updated the comment